python/cuml/dask/preprocessing/LabelEncoder.py

```diff
@@ -0,0 +1,190 @@
+                   DelayedInverseTransformMixin):
+    """
+    An nvcategory based implementation of ordinal label encoding
+
+    Parameters
+    ----------
+    handle_unknown : {'error', 'ignore'}, default='error'
+        Whether to raise an error or ignore if an unknown categorical feature
+        is present during transform (default is to raise). When this parameter
+        is set to 'ignore' and an unknown category is encountered during
+        transform or inverse transform, the resulting encoding will be null.
+
+    Examples
+    --------
+    Converting a categorical implementation to a numerical one
+
+    .. code-block:: python
+
+        import cudf
+        import dask_cudf
+        df = cudf.DataFrame({'num_col':[10, 20, 30, 30, 30],
+                           'cat_col':['a','b','c','a','a']})
+        ddf = dask_cudf.from_cudf(df, npartitions=2)
+
+        # There are two functionally equivalent ways to do this
+        le = LabelEncoder()
+        le.fit(ddf.cat_col)  # le = le.fit(data.category) also works
+        encoded = le.transform(ddf.cat_col)
+
+        print(encoded.compute())
+
+        # This method is preferred
+        le = LabelEncoder()
+        encoded = le.fit_transform(ddf.cat_col)
+
+        print(encoded.compute())
+
+        # We can assign this to a new column
+        ddf = ddf.assign(encoded=encoded)
+        print(ddf.compute())
+
+        # We can also encode more data
+        test_data = cudf.Series(['c', 'a'])
+        encoded = le.transform(dask_cudf.from_cudf(test_data, npartitions=2))
+        print(encoded.compute())
+
+        # After train, ordinal label can be inverse_transform() back to
+        # string labels
+        ord_label = cudf.Series([0, 0, 1, 2, 1])
+        ord_label = dask_cudf.from_cudf(ord_label, npartitions=2)
+
+        print(ord_label.compute())
+
+    Output:
+
+    .. code-block:: python
+
+        [0 1 2 0 0]
+
+        [0 1 2 0 0]
+
+           num_col cat_col  encoded
+        0       10       a        0
+        1       20       b        1
+        2       30       c        2
+        3       30       a        0
+        4       30       a        0
+
+        [2 0]
+
+        0    a
+        1    a
+        2    b
+        0    c
+        1    b
+        dtype: object
+
+    """
+    def __init__(self, client=None, verbose=False, **kwargs):
+        super(LabelEncoder, self).__init__(client=client,
+                                           verbose=verbose,
+                                           **kwargs)
+
+    def fit(self, y):
+        """
+        Fit a LabelEncoder (nvcategory) instance to a set of categories
+
+        Parameters
+        ----------
+        y : dask_cudf.Series
+            Series containing the categories to be encoded. It's elements
+            may or may not be unique
+
+        Returns
+        -------
+        self : LabelEncoder
+            A fitted instance of itself to allow method chaining
```

Comment:
Maybe we should add a "Notes" section that the set of classes will be collected on the client so if you have an enormous number of classes, so it consumes client memory proportional to number of unique items in y? Seems fine for most usecases but good for the user to understand.